scaladoc/src/dotty/tools/scaladoc/tasty/SymOps.scala

```diff
@@ -146,11 +146,19 @@ object SymOps:
 
     def extendedSymbol: Option[reflect.ValDef] =
       import reflect.*
-      Option.when(sym.isExtensionMethod){
-        val termParamss = sym.tree.asInstanceOf[DefDef].termParamss
-        if sym.isLeftAssoc || termParamss.size == 1 then termParamss(0).params(0)
-        else termParamss(1).params(0)
-      }
+      if sym.isExtensionMethod then
+        sym.extendedTermParamLists.find(param => !param.isImplicit && !param.isGiven).flatMap(_.params.headOption)
+      else None
+
+    def splitExtensionParamList: (List[reflect.ParamClause], List[reflect.ParamClause]) =
+      import reflect.*
+      val method = sym.tree.asInstanceOf[DefDef]
+      (for {
+        defPosition <- method.symbol.pos
+        defStart <- scala.util.Try(defPosition.start).toOption
```

Comment:
Does `start` here throw exceptions?

If so, I'm not sure if it's not a bug

Reply:
Yes it does. It fails assertions for some methods.

Reply:
@nicolasstucki is this intentional?

Reply:
What is the exception?

Does the position exist? If not, calling start should fail an assertion. It would be unexpected for the position to not 
exist and I would say that is a bug.

Reply:
It's the assertion failure just as you said.

Reply:
Assertion fails for methods, which have position of type `NoSourcePosition`. These are mostly methods from inheritance such as productElementNames from `Product` trait.  Is this a bug, as `symbol.pos` returns `Option` and logically it should return `None` for `NoSourcePosition` ?